tsconfig.typecheck.json

```diff
@@ -0,0 +1,8 @@
+{
```

Comment:
Please help me to understand this

Reply:
As mentioned in the package.json comment this is so that we can exclude all of the examples from the type checks. As long as we have to make a separate config for that I figured it would be good to add the other type checking settings like skipLibCheck and noEmit. We extend from the base tsconfig to make sure we use the same settings